server/src/main/java/org/apache/druid/client/CachingClusteredClient.java

```diff
@@ -369,6 +373,8 @@ private ClusterQueryResult(Sequence<T> sequence, int numQueryServers)
 
       query = scheduler.prioritizeAndLaneQuery(queryPlus, segmentServers);
       queryPlus = queryPlus.withQuery(query);
+      queryPlus = queryPlus.withQueryMetrics(toolChest);
+      queryPlus.getQueryMetrics().reportQueriedSegmentCount(segmentServers.size()).emit(emitter);
```

Comment:
`RetryQueryRunner` is responsible for re-routing the query to new homes of segments if they are moved during query processing. It uses `CachingClusteredClient` for re-routing. As a result, this can report the same segments multiple times if those segments are moved. I think this is OK and worth documenting.

Reply:
@jihoonson That retry code is unfamiliar to me at this time, `Number of segments that will be touched by the query. If the query has to be retried, the metric will be reported for all retries as well as the original query.` Is that a sane description of the metric in the metrics.md file?